src/Evolution/DiscontinuousGalerkin/Messages/BoundaryMessage.hpp

```diff
@@ -30,6 +31,11 @@ template <size_t Dim>
 struct BoundaryMessage : public CMessage_BoundaryMessage<Dim> {
   using base = CMessage_BoundaryMessage<Dim>;
 
+  // Needed for charm registration
+  static std::string name() {
+    return "BoundaryMessage<" + get_output(Dim) + ">";
```

Comment:
can you just use pretty_type instead of defining your own name?

Reply:
No, because then there's an implicit default. This _needs_ to be uniquely defined for Charm++, so much so that it needs to encode template parameters. `pretty_type` makes it too easy for there to be a subtle bug.

Reply:
No I can't. That was my first instinct too. However, for charm registration in `CharmRegistration.hpp`, I specifically need the `< >` part of the name included which isn't when I use `pretty_type`.